prime-router/src/main/kotlin/azure/observability/AzureCustomDimensionsSerializable.kt

```diff
@@ -0,0 +1,40 @@
+package gov.cdc.prime.router.azure.observability
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ContainerNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import gov.cdc.prime.router.common.JacksonMapperUtilities
+
+/**
+ * Interface for all types that can be serialized and passed to Azure AppInsights
+ */
+interface AzureCustomDimensionsSerializable {
+
+    /**
+     * Takes all fields from a data class and serializes it to a simple String -> String map
+     * that the Microsoft Azure AppInsights Telemetry library can handle.
+     *
+     * This will blow up if the implementing class serializes to something other than an object
+     */
+    fun serialize(): Map<String, String> {
+        return JacksonMapperUtilities.jacksonObjectMapper
+            .valueToTree<ObjectNode>(this)
+            .fields()
+            .asSequence()
+            .associate { it.key to serializeValue(it.value) }
+    }
+
+    /**
+     * All primitives will be in their string format.
```

Comment:
I think this is fine. I looked through the observability doc and didn't see any product metrics that would require us to store numbers here but it would be nice if we could also store numbers to help with future proofing.

Reply:
we can store numbers but it has to be in its string format. If we need to do some kind of calculation on the query side we can parse it back into an int (or long or whatever we need).